internal/core/window.rs

```diff
@@ -220,6 +213,53 @@ pub enum InputMethodRequest {
     },
 }
 
+/// The constraints on a window
+#[non_exhaustive]
+#[derive(Copy, Clone, Debug, PartialEq, Default)]
+pub struct LayoutConstraints {
+    /// Minimum size of the window.
+    pub min: Option<crate::api::LogicalSize>,
+    /// Maximum size of the window.
+    pub max: Option<crate::api::LogicalSize>,
+    /// Preferred size of the window.
+    pub preferred: crate::api::LogicalSize,
+}
+
+/// This struct contains getter that allow to access the window of the property
+/// This is meant to be used in [`WindowAdapter::update_window_properties`].
+pub struct WindowProperties<'a>(&'a WindowInner);
```

Comment:
I wonder if this should instead be in `api.rs`? I think that it is meant to be part of the public API.

Reply:
No, this is in the platform module.
Would you prefer if it was in platform.rs ?

Reply:
And if i don't want to use trick like pub(crate), it should be in this module

Reply:
> No, this is in the platform module.

What's "this"? 

> Would you prefer if it was in platform.rs ?

I don't mind too much in which file it is TBH.

But I'm curious where in the public API it will be. It seems too now it's not anywhere public, or did I miss something?

Reply:
Nevermind, i see the pub use in platform.rs :)